lib/container/ContainerPlugin.js

```diff
@@ -0,0 +1,193 @@
+/*
+	MIT License http://www.opensource.org/licenses/mit-license.php
+	Author Tobias Koppers @sokra, Zackary Jackson @ScriptedAlchemy, Marais Rossouw @maraisr
+*/
+
+"use strict";
+
+const JavascriptModulesPlugin = require("../javascript/JavascriptModulesPlugin");
+const Template = require("../Template");
+const propertyAccess = require("../util/propertyAccess");
+const ContainerExposedDependency = require("./ContainerExposedDependency");
+const ContainerEntryDependency = require("./ContainerEntryDependency");
+const ContainerEntryModuleFactory = require("./ContainerEntryModuleFactory");
+
+const { ConcatSource } = require("webpack-sources");
+const validateOptions = require("schema-utils");
+
+const PLUGIN_NAME = "ContainerPlugin";
+
+module.exports = class ContainerPlugin {
+	constructor(options) {
+		const name = options.name || "remoteEntry";
+
+		validateOptions(
+			{
+				type: "object",
+				properties: {
+					shared: {
+						type: "object"
+					},
+					expose: {
+						type: ["object", "array"]
+					},
+					name: {
+						type: "string",
+						default: name
+					},
+					library: {
+						type: "string",
+						default: name
+					},
+					libraryTarget: {
+						type: "string",
+						default: "global",
+						enum: [
+							"var",
+							"this",
+							"window",
+							"self",
+							"global",
+							"commonjs",
+							"commonjs2",
+							"amd",
+							"amd-require",
+							"umd",
+							"umd2",
+							"system"
+						]
+					},
+					filename: {
+						anyOf: [{ type: "string" }, { instanceof: "Function" }]
+					}
+				},
+				additionalProperties: false
+			},
+			options,
+			{ name: PLUGIN_NAME }
+		);
+
+		this.options = {
+			shared: options.shared || null,
+			name,
+			library: options.library || name,
+			libraryTarget: options.libraryTarget || "global",
+			filename: options.filename || undefined, // Undefined means, use the default behaviour
+			expose: options.expose || {}
```

Comment:
Use the `parseOptions` method

Reply:
Maybe rename it to `exposes`? It would be consistent with `overridables` and `overrides`